src/debug/utils.ts

```diff
@@ -76,10 +77,18 @@ export class PackageMap {
 					this.map[name] = uriToFilePath(rest);
 				else
 					this.map[name] = path.join(path.dirname(file), rest);
+
+				// If we map to "lib/" then this must be the local package so we can stash the name.
+				if (rest == "lib/")
```

Comment:
I thought that at first, but then I figured that while the `.packages` file exists to provide a mapping from `packages:xxx` uris to actual locations, it'll always have to be there?

Unless you mean specifically being `lib/`.. I guess it could reasonably become a full path and then this would break.

Another option is reading it from `pubspec.yaml`; I just went with this as we already had read the data. Do you think that would be better? (though maybe I need to think about/test without a `pubspec.yaml`/`.packages` too, I presume it's totally valid to want to debug just a single .dart file in a folder?